dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/PersistablesFilter.java

```diff
@@ -171,16 +171,9 @@ private void filter() {
     }
   }
 
-  private <T extends TrackerDto> void collectDeletables(
-      Class<T> type, List<Function<T, TrackerDto>> parents, List<T> entities) {
+  private <T extends TrackerDto> void collectDeletables(Class<T> type, List<T> entities) {
     for (T entity : entities) {
-      if (isValid(entity) && isDeletable(entity)) {
```

Comment:
Nice catch, that was a tough one.
The test was covering the validation part but only because the strategy atomic mode by default is `ALL` and if there is an error, no object is persisted.
I updated the test to import one existing and one not existing relationship with atomic mode `OBJECT` and verifying that in the report there is one error and one relationship was deleted